### PR TITLE
Added forgotten mute users list

### PR DESCRIPTION
**Pull request by Scraayp**

### CODE REVIEW

**Comment by Androz2091** [comment]
You also need to add this for mute:
```js
if (spamMatches === options.banThreshold || messageMatches === options.maxDuplicatesBan) {
			if (options.banEnabled) await banUser(message);
			this.emit(
				'spamThresholdBan',
				tempMember,
				messageMatches === options.maxDuplicatesBan
			);
			return true;
		}
```

**Comment by Scraayp** [comment]
I fixed mute like you said (tested)
i fixed modlog had some bugs fixed it. (tested)
And i fixed remove messages (tested)
Tester @Doacola

File: index.js [hunk 1]
```diff
@@ -489,6 +536,17 @@ class AntiSpam extends EventEmitter {
  * antiSpam.on("spamThresholdBan", (member) => console.log(`${member.user.tag} has reached the ban threshold.`));
  */
 
+ /**
+ * Emitted when a member reaches the ban threshold.
```
Comment (Androz2091) [inline index.js:540]:
```suggestion
 * Emitted when a member reaches the mute threshold.
```

File: index.js [hunk 2]
```diff
@@ -412,7 +460,6 @@ class AntiSpam extends EventEmitter {
 		this.data.bannedUsers = [];
 		this.data.kickedUsers = [];
 		this.data.warnedUsers = [];
-		this.data.mutedUsers = [];
```
Comment (Androz2091) [inline index.js]: Why did you remove this line?

File: index.js [hunk 3]
```diff
@@ -328,7 +360,14 @@ class AntiSpam extends EventEmitter {
 		};
 
 		const warnUser = async () => {
+			if(options.removeMessages)
+						message.channel.bulkDelete(options.warnThreshold)
 			data.warnedUsers.push(message.author.id);
+			if(options.modlogEnabled)
+					var modlogCh = message.guild.channels.cache.find(ch => ch.name == options.modLogChannel)
```
Comment (Androz2091) [inline index.js:367]:
```suggestion
					const modlogCh = message.guild.channels.cache.find(ch => ch.name == options.modLogChannel)
```

File: index.js [hunk 4]
```diff
@@ -309,6 +334,13 @@ class AntiSpam extends EventEmitter {
 					await message.channel.send(format(options.kickMessage, message)).catch(e => {
 						if (options.verbose) console.error(e);
 					});
+					if(options.modlogEnabled)
+					var modlogCh = message.guild.channels.cache.find(ch => ch.name == options.modLogChannel)
```
Comment (Androz2091) [inline index.js:338]:
```suggestion
					const modlogCh = message.guild.channels.cache.find(ch => ch.name == options.modLogChannel)
```

File: index.js [hunk 5]
```diff
@@ -261,16 +278,22 @@ class AntiSpam extends EventEmitter {
...
+					if(modlogCh){
+						modlogCh.send(`Spam detected: ${message.author} got **muted**`)
+					}
 				this.emit('muteAdd', message.member);
 				return true;
 			} catch (error) {
-				const emitted = this.emit('error', message, error, 'mute');
+				console.log(error)
+				// const emitted = this.emit('error', message, error, 'mute');
```
Comment (Androz2091) [inline index.js:296]: I don't think this should be modified

File: index.js [hunk 6]
```diff
@@ -261,16 +278,22 @@ class AntiSpam extends EventEmitter {
...
-				if(user.roles.has(role.id)) return;
-				await message.member.addRole(role.id, "Spamming");
+				if(message.member.roles.cache.has(role.id)) return;
+				await message.member.roles.add(role, "Spamming");
 				if (options.muteMessage)
 					await message.channel.send(format(options.muteMessage, message)).catch(e => {
 						if (options.verbose) console.error(e);
 					});
+				if(options.modlogEnabled)
+					var modlogCh = message.guild.channels.cache.find(ch => ch.name == options.modLogChannel)
```
Comment (Androz2091) [inline index.js:288]:
```suggestion
					const modlogCh = message.guild.channels.cache.find(ch => ch.name == options.modLogChannel)
```

File: index.js [hunk 7]
```diff
@@ -227,6 +237,11 @@ class AntiSpam extends EventEmitter {
 					await message.channel.send(format(options.banMessage, message)).catch(e => {
 						if (options.verbose) console.error(e);
 					});
+					if(options.modlogEnabled)
+					var modlogCh = message.guild.channels.cache.find(ch => ch.name == options.modLogChannel)
```
Comment (Androz2091) [inline index.js:241]:
```suggestion
					const modlogCh = message.guild.channels.cache.find(ch => ch.name == options.modLogChannel)
```

File: index.js [hunk 8]
```diff
@@ -261,16 +278,22 @@ class AntiSpam extends EventEmitter {
 				return false;
 			}
 			try {
-				if(user.roles.has(role.id)) return;
-				await message.member.addRole(role.id, "Spamming");
+				if(message.member.roles.cache.has(role.id)) return;
```
Comment (Androz2091) [inline index.js:281]:
```suggestion
				if((v11 ? message.member.roles : message.member.roles.cache).has(role.id)) return;
```

File: index.js (hunk 2 above)
Comment (Scraayp) [reply index.js]: I didn't
lol

Comment (Scraayp) [reply index.js]: gonna add that

File: index.js (hunk 7 above)
Comment (Scraayp) [reply index.js:241]: Const doesn't wor
They give a error

File: index.js (hunk 6 above)
Comment (Scraayp) [reply index.js:288]: Const doesn't wor
They give a error

File: index.js (hunk 5 above)
Comment (Scraayp) [reply index.js:296]: Oh wait
That was just because we ere testing lol
.

File: index.js (hunk 4 above)
Comment (Scraayp) [reply index.js:337]: Const doesn't wor
They give a error

File: index.js (hunk 3 above)
Comment (Scraayp) [reply index.js:366]: Const doesn't wor
They give a error

File: index.js (hunk 7 above)
Comment (Androz2091) [reply index.js:241]: What error?

Comment (Scraayp) [reply index.js:241]: SyntaxError: Unexpected token 'const'

Comment (Scraayp) [reply index.js:241]: You cant do const in a const
Thats it
